apps/emqx_ft/src/emqx_ft.erl

```diff
@@ -145,10 +146,11 @@ on_file_command(PacketId, Msg, FileCommand) ->
     case string:split(FileCommand, <<"/">>, all) of
         [FileId, <<"init">>] ->
             on_init(PacketId, Msg, transfer(Msg, FileId));
-        [FileId, <<"fin">>] ->
-            on_fin(PacketId, Msg, transfer(Msg, FileId), undefined);
-        [FileId, <<"fin">>, Checksum] ->
-            on_fin(PacketId, Msg, transfer(Msg, FileId), Checksum);
+        [FileId, <<"fin">>, FinalSizeBin | ChecksumL] ->
+            validate([{size, FinalSizeBin}], fun([FinalSize]) ->
+                Checksum = emqx_maybe:from_list(ChecksumL),
+                on_fin(PacketId, Msg, transfer(Msg, FileId), FinalSize, Checksum)
```

Comment:
Don't we have is a small inconsistency? `on_fin` receives unparsed and unvalidated checksum, while `on_segment` receives parsed and validated one.

Reply:
Attempted to fix with 522cb5fb377f3bf370bb8ab2493c095afbd3732c.